### PR TITLE
fix(Filter): Destroy the webgl context used for testing

### PR DESCRIPTION
**Pull request by asturur**
## Motivation

This was brought up in #8922 and is being backported to master.

### CODE REVIEW

**Review by ShaMan123 — APPROVED** [review]

**Comment by ShaMan123** [comment]
It is kind of strange it isn't lost be itself

**Review by ShaMan123 — COMMENTED** [review]
It is kind of strange it isn't lost by itself
We don't hold a ref to it
Maybe we should call `dispose` on the canvas as well...

**Comment by asturur** [comment]
the canvas is a plain HTMLElement one, and is Garbage collected, but is easy to try, i will make a function that call this one in rapid succession to see what happens.

Not a priority for me

**Comment by asturur** [comment]
I don't see either firefox or chrome crashing with current master even in a tight loop executing that function.
There is some cleanup to do tho.

The benchmark code should go away from initialization since is very heavy and should become an utility for the developer if he want to check.

Can't be a forced check for everyone.

File: src/filters/utils.ts [hunk 1]
```diff
@@ -1,7 +1,54 @@
...
+  startTime = getFabricWindow().performance.now();
+  WebGLFilterBackend.prototype.copyGLTo2DPutImageData.call(
+    testContext,
+    gl,
+    testPipelineState
+  );
+  const putImageDataTime = getFabricWindow().performance.now() - startTime;
+
+  return drawImageTime > putImageDataTime;
+};
```
[the remark this answers is not in the document]
Comment (ShaMan123) [reply src/filters/utils.ts:54]: This is really interesting
I wasn't aware it existed

File: src/filters/utils.ts [hunk 2]
```diff
@@ -1,7 +1,54 @@
...
+    testPipelineState
+  );
+  const drawImageTime = getFabricWindow().performance.now() - startTime;
+
+  startTime = getFabricWindow().performance.now();
+  WebGLFilterBackend.prototype.copyGLTo2DPutImageData.call(
+    testContext,
+    gl,
+    testPipelineState
+  );
```
Comment (ShaMan123) [inline src/filters/utils.ts:50]: interested why not use an instance instead of calling 2 proto methods

**Pull request merged**